java-checks/src/main/java/org/sonar/java/checks/spring/StatusCodesOnResponseCheck.java

```diff
@@ -217,12 +216,21 @@ private Tree checkCatch(MethodInvocationTree methodInvocationTree, boolean isErr
     private Tree checkTry(MethodInvocationTree methodInvocationTree, boolean isOk) {
       Tree tryParent = ExpressionUtils.getParentOfType(methodInvocationTree, Tree.Kind.TRY_STATEMENT);
 
-      if (tryParent != null && !isOk) {
-        reportIssue(methodInvocationTree, ISSUE_MESSAGE);
+      if (tryParent != null) {
+        Tree ifParent = ExpressionUtils.getParentOfType(methodInvocationTree, Tree.Kind.IF_STATEMENT);
```

Comment:
Shouldn't we look for the `ifParent` of the the `tryParent`?
```suggestion
        Tree ifParent = ExpressionUtils.getParentOfType(tryParent, Tree.Kind.IF_STATEMENT);
```